cmake/modules/BuildBoost.cmake

```diff
@@ -171,6 +171,7 @@ function(do_build_boost version)
   include(ExternalProject)
   ExternalProject_Add(Boost
     ${source_dir}
+    INSTALL_DIR "external"
```

Comment:
why?

Reply:
I tried keeping it build only, but jaeger dependencies needed Boost installed files during build time, distro package boost lead to interfare with Ceph build boost, so I instead went with installing Ceph Boost altogether.

Reply:
which jaeger dependencies need Boost installed files during build time in what way?

Reply:
thrift does and when I did build alone, i still can see include files generated, but when I tried compiling without install I am getting: https://github.com/ceph/ceph/pull/31358/files#r513359062 locally no issues seen in build so I suppose it's with the package time path configuration, I will test it today and check what could be done